tests/integration_tests.rs

```diff
@@ -589,37 +597,49 @@ fn do_not_exit_directory() {
 }
 
 #[test]
+#[serial]
 fn pager_basic() {
-    bat()
-        .env("PAGER", "echo pager-output")
-        .arg("--paging=always")
-        .arg("test.txt")
-        .assert()
-        .success()
-        .stdout(predicate::eq("pager-output\n").normalize());
+    mocked_pagers::with_mocked_versions_of_more_and_most_in_path(|| {
```

Comment:
Thank you for your contribution. I don't quite follow along here though. Why do we need to change for example this test?

Reply:
You're welcome. The reason we're using the mocked versions here, is so that we actually have an `echo` available for powershell. See my comment on the original bug report: https://github.com/sharkdp/bat/issues/2592. `echo` is not resolved, so on powershell tests that rely on it fail.

Reply:
That makes sense. I never tried to run `cargo test` in PowerShell on Windows before. But now when I do these tests indeed fail. Any chance you would be willing to look into running our tests in CI in PowerShell on Windows? Some instructions: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-powershell

In our `.github\workflows\CICD.yml` and `matrix:` I think we would want to parameterize on `shell:` so that Windows tests run with `shell: pwsh`. Unless we continuously run the tests in PowerShell, it will for sure break again without anyone not noticing.

If you are not interested in doing this, that's totally fine. Just let me know, and I will do a more detailed review of this PR as it is.

Reply:
Sure, I don't mind taking a look. Do you want it as part of this PR?

Reply:
Great! Yes I think that would be good, that way we know the tests pass in CI PowerShell before we merge.

Reply:
Made a slight change to the test step so that tests are ran in powershell on windows targets. However, it seems a new test introduced with a merge on `master` yesterday seems to be failing intermittently. The test is `lessopen_and_lessclose_stdin_temp`.
It's also failing in other pipelines (see https://github.com/sharkdp/bat/actions/runs/6059140072), and I can reproduce it locally by running `cargo test` a few times on `master`.

Reply:
@Anomalocaridid Hello! Are you by any chance able to look into the reported flakiness of `lessopen_and_lessclose_stdin_temp`? Would be great if you could 🙏

Reply:
@Enselic After taking a quick look at the test output and code, I am not entirely sure what the cause is.

Also, strangely, if I only run the integration tests with `cargo test --test integration_tests` it does not appear to fail on my system. Even when running it repeatedly using `chronic` from moreutils.

If I had to make an educated guess, it's some sort of data race that's interfering with `$LESSCLOSE`. Likely by either keeping it from executing or by just preventing its output from reaching the test runner. I am not sure if the stdin and temp file parts of the specific test are to blame or if it's somehow related to what order `cargo` runs the tests in.

I'll try to take a closer look later when I get a chance, but I am honestly not even sure how to go from here.

Reply:
I can't reproduce any flakiness unfortunately. Not in WSL 2 nor in PowerShell.

@boyvanduuren Feel free to add an `#[ignore = "failing intermittently on some systems"]` in your PR for now.

Reply:
All green.

Reply:
@boyvanduuren Thanks

@Anomalocaridid: if I run that test "manually", I get an error message from `less`:
```
> export LESSOPEN="-echo empty.txt"
> export LESSCLOSE="echo lessclose: %s %s"
> echo "test.txt" | bat
LESSOPEN ignored: must contain exactly one %s
```

Could this be the issue? Maybe the race condition is somehow related to whether or not that error is printed first.

Reply:
@sharkdp Possibly? However, it looks like that error also appears when manually running the other tests where `$LESSCLOSE` has a dash as part of its prefix. But those do not have the same issue, right?

A [recent comment](https://github.com/sharkdp/bat/pull/2444#issuecomment-1704811314) on my PR for `$LESSOPEN` support gave me a helpful lead.

Basically, they said that the files were going through `$LESSOPEN` twice, through both `bat` and `less`. Sure enough, by running the test manually with `less` instead of bat, the same `LESSOPEN ignored` message appears. Also it may be worth mentioning that the message is not present in the code for bat's `$LESSOPEN` implementation.

So for some reason, the `$LESSOPEN` environment variable is being passed on to `less`, but only when `$LESSOPEN` has a dash in it that indicates that stdin is to be preprocessed as well as files.

When I get a chance, I'll try to fix both `$LESSOPEN` being passed to `less` as well as make `bat` more closely match `less`'s behavior for `$LESSOPEN` without a `%s`.